File: StrategyPatternJava/MallardDuck.java
public class MallardDuck extends Duck {
    public MallardDuck(){
        setFlyBehavior(new FlyWithWings());
        setQuackBehavior(new Quack());
    }

    public void display(){
        System.out.println("I am a Mallard Duck");
    }
}

File: StrategyPatternJS/MallardDuck.js
let Duck = require("./Duck");
let FlyWithWings = require("./FlyWithWings");
let Quack = require("./Quack");

module.exports = class MallardDuck extends Duck {
    constructor(){
        super();
        this.fliesBy = new FlyWithWings();
        this.quacksBy = new Quack();
    }

    display() {
        console.log("I'm a Mallard Duck");
    }
};